File: mobile/ios/Quote/Features/Auth/Models/AuthModels.swift
import Foundation

struct LoginRequest: Encodable {
    let email: String
    let password: String
}

struct RegisterRequest: Encodable {
    let email: String
    let password: String
    let firstName: String
    let lastName: String
    let phone: String?
    let userType: Int
    let abn: String?
}

struct AuthResponse: Decodable {
    let userId: String
    let email: String
    let firstName: String?
    let lastName: String?
    let userType: String?
    let accessToken: String
    let refreshToken: String
}

struct User: Codable, Identifiable {
    let id: String
    let email: String
    let firstName: String
    let lastName: String
    let userType: UserType
    let profilePhotoUrl: String?

    var fullName: String {
        "\(firstName) \(lastName)"
    }
}

enum UserType: String, Codable {
    case customer = "Customer"
    case tradie = "Tradie"
    case admin = "Admin"
}

File: mobile/android/app/src/main/java/com/quote/app/features/auth/repository/AuthModels.kt
package com.quote.app.features.auth.repository

data class LoginRequest(
    val email: String,
    val password: String
)

data class RegisterRequest(
    val email: String,
    val password: String,
    val firstName: String,
    val lastName: String,
    val phone: String?,
    val userType: Int,
    val abn: String?
)

data class AuthResponse(
    val userId: String,
    val email: String,
    val firstName: String?,
    val lastName: String?,
    val userType: String?,
    val accessToken: String,
    val refreshToken: String
)
